### ISSUE TITLE
Custom metrics not avialble in metric alerts

### ISSUE THREAD

**Issue by JLuse** [getsentry/sentry#55159]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

**Description:**
When adding a custom metric, you are unable to select it as an option in metric alerts.

Custom metric:
<img width="1018" alt="Screenshot 2023-08-21 at 3 10 01 PM" src="https://github.com/getsentry/sentry/assets/8451987/8ba48afd-73bb-4d47-b77d-116c162fa3cc">

Alert options:
<img width="283" alt="Screenshot 2023-08-21 at 3 13 51 PM" src="https://github.com/getsentry/sentry/assets/8451987/13f1bbf4-5edc-400f-af08-8f041a326b8b">

Steps:
1. Send custom metric with a transaction.
2. Create alert -> select Custom Metric
3. Attempt to select metric from dropdown

### Expected Result

Custom metric to be utilized in alert

### Actual Result

Unable to select custom measurements besides the predefined options

### Product Area

Alerts

**Comment by nanoburd** [comment]
@JLuse what was the use case the customer was trying to do here?

**Comment by guiquintelas** [comment]
@nanoburd 
> @JLuse what was the use case the customer was trying to do here?

Hey, customer here! The goal is to create alerting based on the `custom.waiting_time` custom measurement and many others. We're trying to use Sentry to monitor our internal job queue. So we have many metrics that don't fit into the existing ones.

Another use case is to simply use custom metrics as a way to filter down our transactions. As tags do not currently work to do so. 

We currently have:
![image](https://github.com/getsentry/sentry/assets/29166076/0b4e046a-b46f-49ef-ba34-5ee58243787c)

Solving this issue allows us to finally stop jamming data into our transaction names:
![image](https://github.com/getsentry/sentry/assets/29166076/7f4a10c3-dabf-4581-abea-4334b3f1303a)

**Comment by brentc** [comment]
Thanks for this additional context. I'm adding this to our backlog for now and we'll provide an update when we have more information.

**Comment by nanoburd** [comment]
Yup this is likely something we can work on, I think this is valuable. Thanks for bringing this up!~

**Comment by realkosty** [comment]
if we don't fix this in triage party we should at least change UI that makes it look like we support it

<img width="442" alt="Image" src="https://github.com/user-attachments/assets/d2571b6f-10a7-4b88-ab6c-43ac16a6ef70" />

**Comment by cobyeastwood183** [comment]
https://sentry.zendesk.com/agent/tickets/147061

**Comment by conradkoh-fp** [comment]
I found something interesting on this. Even though it is not working when we do `Alerts > Custom Measurement`, it seems to show when we create a dashboard, view the metric, choose "Create Alert".

Should this fall in the category of a bug then?
